src/main/java/com/cleanroommc/groovyscript/compat/mods/thaumcraft/aspect/AspectHelper.java

```diff
@@ -0,0 +1,335 @@
+                    this.add(aspectList.entity, as, false);
+        });
+    }
+
+    @GroovyBlacklist
+    public void addScripted(Object target, AspectStack aspect) {
+        AtomicBoolean found = new AtomicBoolean(false);
+        scripted.forEach(scriptedAspect -> {
+            if (target instanceof EntityEntry && scriptedAspect.entity != null
+                    && ((EntityEntry) target).getName().equals(scriptedAspect.entity.getName())) {
+                found.set(true);
+                scriptedAspect.addAspect(aspect);
+            } else if (target instanceof ItemStack && scriptedAspect.item != null
+                    && ((ItemStack) target).isItemEqual(scriptedAspect.item)) {
+                found.set(true);
+                scriptedAspect.addAspect(aspect);
+            }
+        });
+
+        if (!found.get()) {
+            ArrayList<AspectStack> aspectList = new ArrayList<>();
+            aspectList.add(aspect);
+            if (target instanceof ItemStack)
+                scripted.add(new AspectListHelper((ItemStack) target, aspectList));
+            else if (target instanceof EntityEntry)
+                scripted.add(new AspectListHelper((EntityEntry) target, aspectList));
+        }
+    }
+
+    @GroovyBlacklist
+    public void addBackup(Object target, AspectStack aspect) {
+        AtomicBoolean found = new AtomicBoolean(false);
+        backup.forEach(backupAspect -> {
+            if (target instanceof EntityEntry && backupAspect.entity != null
+                    && ((EntityEntry) target).getName().equals(backupAspect.entity.getName())) {
+                found.set(true);
+                backupAspect.addAspect(aspect);
+            } else if (target instanceof ItemStack && backupAspect.item != null
+                    && ((ItemStack) target).isItemEqual(backupAspect.item)) {
+                found.set(true);
+                backupAspect.addAspect(aspect);
+            }
+        });
+
+        if (!found.get()) {
+            ArrayList<AspectStack> aspectList = new ArrayList<>();
+            aspectList.add(aspect);
+            if (target instanceof ItemStack)
+                backup.add(new AspectListHelper((ItemStack) target, aspectList));
+            else if (target instanceof EntityEntry)
+                backup.add(new AspectListHelper((EntityEntry) target, aspectList));
+        }
+    }
+
+    public void add(EntityEntry entity, AspectStack aspect) { this.add(entity, aspect, true); }
+    public void add(OreDictIngredient oreDict, AspectStack aspect) { this.add(oreDict, aspect, true); }
+    public void add(ItemStack item, AspectStack aspect) { this.add(item, aspect, true); }
+
+    public void remove(EntityEntry entity, AspectStack aspect) { this.remove(entity, aspect, true); }
+    public void remove(OreDictIngredient oreDict, AspectStack aspect) { this.remove(oreDict, aspect, true); }
+    public void remove(ItemStack item, AspectStack aspect) { this.remove(item, aspect, true); }
```

Comment:
Please format this properly

Reply:
fixed in commit: [d242a72](https://github.com/CleanroomMC/GroovyScript/pull/18/commits/d242a72a6c5de01ec2ae925403af44a157d14bb3)